ledger-tool/src/bigtable.rs

```diff
@@ -308,6 +337,14 @@ impl BigTableSubCommand for App<'_, '_> {
                 .about("Ledger data on a BigTable instance")
                 .setting(AppSettings::InferSubcommands)
                 .setting(AppSettings::SubcommandRequiredElseHelp)
+                .arg(
+                    Arg::with_name("bigtable_instance")
+                        .long("bigtable-instance")
+                        .global(true)
```

Comment:
Did you test this out?  I think there are `Arg::global(true).default_value("default")` bugs.  I wasn't able to get both
```
solana-ledger-tool bigtable --bigtable-instance  custom-name <SUBCOMMAND>
```
and
```
solana-ledger-tool bigtable <SUBCOMMAND> --bigtable-instance  custom-name
```

to work in any sane way. my last shot was going to be testing both `matches` and `sub_matches` for presence and the default value and take whichever was non-default should that be the case. :face_with_head_bandage:

Reply:
Oh dang, I didn't realize I'd broken case 2. This alternative seems to work, though...

```
fn bigtable_subcommand(self) -> Self {
        self.arg(
            Arg::with_name("bigtable_instance")
                .long("bigtable-instance")
                .global(true)
                .takes_value(true)
                .default_value("solana-ledger")
                .help("Name of BigTable instance to target"),
        )
        .subcommand(
            ...
```
Wdyt?
This will technically allow `--bigtable-instance` before the `bigtable` subcommand too, which I think is okay.

Reply:
I was hoping to keep the arg out of the top-level help, but this might be the least bad option

Reply:
I pushed it, for mulling purposes.
Want me to incorporate any of your changes to other parts of this set, though?

Reply:
sorry for the delay!

i've pushed mine [here](https://github.com/solana-labs/solana/pull/23785).  i think i prefer it, but with your first and third cherry-picked on top.  i could be convinced that your fourth is better than my giant, comment-worthy block of stupid, but exposing the arg in irrelevant contexts rubs me wrong. wdyt?

Reply:
I thought I wasn't going to like the block of stupid, but I am actually fine with it.
I'm not sure which way will be easier to cherry-pick, but I've got a couple hours this evening to work on it, so I'll either pull your code into mine or push to your branch.

Reply:
Pushed here. How does it look?